Homework_4/numpy_challenge.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[84]:
+
+
+import numpy as np
+
+
+def matrix_multiplication(matrix1, matrix2):
+    '''Function matrix_multiplication() performs matrix multiplication of two 2D matrices
+    
+    Parameters:
+    matrix1(m x n), matrix2(n x k) (numpy.ndarray): input arrays
+    
+    Returns: 
+    matrix(m x k), numpy.array
+    '''
+    
+    return np.matmul(matrix1, matrix2)
+
+
+def multiplication_check(matrix_list):
+    '''Function multiplication_check() determines whether matrices in list can be multiplied in given order
+    
+    Parameters:
+    matrix_list (list): list of numpy.arrays
+    
+    Returns: bools after multiplication check
+    True - matrices in list can be multiplied in given order    
+    '''
+    
+    for index in range(len(matrix_list) - 1):
+        if matrix_list[index].shape[1] != matrix_list[index+1].shape[0]:
+            return False
+        return True
```

Comment:
По тестам была ошибка и я долго не мог понять в чём именно дело...

У тебя `return True` находится внутри цикла, то есть выполнение твоей функции заканчивается сразу после первой итерации, что приводит к неверным результатам
```suggestion
    for index in range(len(matrix_list) - 1):
        if matrix_list[index].shape[1] != matrix_list[index+1].shape[0]:
            return False
    return True
```